demo_nodes_cpp/src/services/add_two_ints_client_async.cpp

```diff
@@ -17,33 +17,38 @@
 #include <iostream>
 #include <memory>
 #include <string>
+#include <vector>
 
 #include "rclcpp/rclcpp.hpp"
+#include "rclcpp_components/register_node_macro.hpp"
 #include "rcutils/cmdline_parser.h"
 
 #include "example_interfaces/srv/add_two_ints.hpp"
 
 using namespace std::chrono_literals;
 
-void print_usage()
+namespace demo_nodes_cpp
 {
-  printf("Usage for add_two_ints_client app:\n");
-  printf("add_two_ints_client [-s service_name] [-h]\n");
-  printf("options:\n");
-  printf("-h : Print this help function.\n");
-  printf("-s service_name : Specify the service name for this client. Defaults to add_two_ints.\n");
-}
-
 class ClientNode : public rclcpp::Node
 {
 public:
-  explicit ClientNode(const std::string & service_name)
-  : Node("add_two_ints_client")
+  explicit ClientNode(const rclcpp::NodeOptions & options)
+  : Node("add_two_ints_client", options)
   {
-    client_ = create_client<example_interfaces::srv::AddTwoInts>(service_name);
-
+    setvbuf(stdout, NULL, _IONBF, BUFSIZ);
+    std::vector<std::string> args = options.arguments();
+    if (find_command_option(args, "-h")) {
+      print_usage();
+      rclcpp::shutdown();
+    } else {
+      std::string tmptopic = get_command_option(args, "-s");
```

Comment:
Looks like the CLI args are only used to set the service name, which can be accomplished with a remap instead. Therefore, I think we can remove all the logic related to parsing CLI args.
But, if we still plan to backport these changes then I would leave it as-is and remove it in a follow-up.